pkg/runtime/runtime-manifest_test.go

```diff
@@ -6,104 +6,92 @@ import (
-			"description": "a test step",
-			"Parameters": map[string]interface{}{
-				"Thing": "{{bundle.parameters.person}}",
-			},
-		},
-	}
+func TestResolveMapParam(t *testing.T) {
+	pCtx := portercontext.NewTestContext(t)
+	pCtx.Setenv("PERSON", "Ralpha")
+
+	mContent := `name: mybuns
+parameters:
+- name: person
+- name: place
+  applyTo: [install]
+
+install:
+- mymixin:
+    Parameters:
+      Thing: "{{ bundle.parameters.person }}"
+`
+	rm := runtimeManifestFromStepYaml(t, pCtx, mContent)
+	s := rm.Install[0]
 
-	before, _ := yaml.Marshal(s)
-	t.Logf("Before:\n %s", before)
-	err := rm.ResolveStep(s)
+	err := rm.ResolveStep(0, s)
 	require.NoError(t, err)
-	after, _ := yaml.Marshal(s)
-	t.Logf("After:\n %s", after)
-	assert.NotNil(t, s.Data)
-	t.Logf("Length of data:%d", len(s.Data))
-	assert.NotEmpty(t, s.Data["Parameters"])
-	for k, v := range s.Data {
-		t.Logf("Key %s, value: %s, type: %T", k, v, v)
-	}
-	pms, ok := s.Data["Parameters"].(map[string]interface{})
-	assert.True(t, ok)
+
+	require.IsType(t, map[string]interface{}{}, s.Data["mymixin"])
+	mixin, ok := s.Data["mymixin"].(map[string]interface{})
+	require.True(t, ok)
+
+	require.IsType(t, mixin["Parameters"], map[string]interface{}{})
+	pms, ok := mixin["Parameters"].(map[string]interface{})
+	require.True(t, ok)
 	val, ok := pms["Thing"].(string)
-	assert.True(t, ok)
+	require.True(t, ok)
 	assert.Equal(t, "Ralpha", val)
 	assert.NotContains(t, "place", pms, "parameters that don't apply to the current action should not be resolved")
 
 	err = rm.Initialize()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 }
 
 func TestResolvePathParam(t *testing.T) {
-	cxt := portercontext.NewTestContext(t)
-	m := &manifest.Manifest{
-		Parameters: manifest.ParameterDefinitions{
-			"person": {
-				Name: "person",
-				Destination: manifest.Location{
-					Path: "person.txt",
-				},
-			},
-		},
-	}
-	rm := NewRuntimeManifest(cxt.Context, cnab.ActionInstall, m)
-	s := &manifest.Step{
-		Data: map[string]interface{}{
-			"description": "a test step",
-			"Parameters": map[string]interface{}{
-				"Thing": "{{bundle.parameters.person}}",
-			},
-		},
-	}
+	pCtx := portercontext.NewTestContext(t)
+
+	mContent := `name: mybuns
+parameters:
+- name: person
+  path: person.txt
+
+install:
+- mymixin:
+    Parameters:
+      Thing: "{{ bundle.parameters.person }}"
+`
+	rm := runtimeManifestFromStepYaml(t, pCtx, mContent)
+	s := rm.Install[0]
 
-	before, _ := yaml.Marshal(s)
-	t.Logf("Before:\n %s", before)
-	err := rm.ResolveStep(s)
+	err := rm.ResolveStep(0, s)
 	require.NoError(t, err)
-	after, _ := yaml.Marshal(s)
-	t.Logf("After:\n %s", after)
-	assert.NotNil(t, s.Data)
-	t.Logf("Length of data:%d", len(s.Data))
-	assert.NotEmpty(t, s.Data["Parameters"])
-	for k, v := range s.Data {
-		t.Logf("Key %s, value: %s, type: %T", k, v, v)
-	}
-	pms, ok := s.Data["Parameters"].(map[string]interface{})
-	assert.True(t, ok)
+
+	require.IsType(t, s.Data["mymixin"], map[string]interface{}{})
```

Comment:
Why do we need to check for a type and then do type assertion again?

Reply:
I have pushed a new commit that finishes cleaning up how we check for types in the unit test.

Using a cast and checking OK like this isn't great because when the test fails, it just says "false is not true". Using IsType with a error message makes it more clear in the test output what failed and why.